src/rai_bench/rai_bench/tool_calling_agent/tasks/custom_interfaces.py

```diff
@@ -158,328 +199,865 @@ def available_tools(self) -> List[BaseTool]:
+                        f"detections.{i}.bbox.center.y": defaults["bbox_center"][1],
+                        f"detections.{i}.bbox.size_x": defaults["bbox_size"][0],
+                        f"detections.{i}.bbox.size_y": defaults["bbox_size"][1],
+                    }
+                )
+
+            pub_detection_array_subtask = CheckTopicFieldsToolCallSubTask(
+                expected_tool_name="publish_ros2_message",
+                expected_topic=DETECTIONS_TOPIC,
+                expected_message_type=DETECTION_ARRAY_MESSAGE_TYPE,
+                expected_fields=expected_fields,
+            )
+
+            call_vector_store_subtask = CheckServiceFieldsToolCallSubTask(
+                expected_tool_name="call_ros2_service",
+                expected_service=VECTOR_STORE_SERVICE,
+                expected_service_type=VECTOR_STORE_SERVICE_TYPE,
+                expected_fields={
+                    "query": self.safety_query,
+                },
+            )
+
+            pub_hri_documentation_subtask = CheckTopicFieldsToolCallSubTask(
+                expected_tool_name="publish_ros2_message",
+                expected_topic=HRI_TOPIC,
+                expected_message_type=HRI_MESSAGE_TYPE,
+                expected_fields={"text": self.doc_report},
+            )
+
+            validators = [
+                OrderedCallsValidator(
+                    subtasks=[
+                        call_what_i_see_subtask,
+                        pub_detection_array_subtask,
+                        call_vector_store_subtask,
+                        pub_hri_documentation_subtask,
+                    ]
+                )
+            ]
+
+        super().__init__(validators=validators, task_args=task_args, logger=logger)
+
+    @property
+    def optional_tool_calls_number(self) -> int:
+        # list services and get interface for all required services
+        return 5
 
     def get_base_prompt(self) -> str:
-        return f"Call service '{self.service}' to get visual observations."
+        object_summary = ", ".join(
+            [
+                f"{obj} at ({center[0]}, {center[1]}) and size {size[0]}x{size[1]}"
+                for obj, center, size in zip(
+                    self.objects, self.bbox_centers, self.bbox_sizes
+                )
+            ]
+        )
 
-    def get_prompt(self) -> str:
-        if self.prompt_detail == "brief":
-            return self.get_base_prompt()
-        else:
-            return (
-                f"{self.get_base_prompt()} "
-                "You can discover available vision services, examine the service "
-                "interface, and call the service with an empty request to get "
-                "visual observations and camera pose information."
+        return f"""Perform comprehensive scene documentation using multiple services:
+            1) Call service '/rai_whatisee_get' to get visual observations
```

Comment:
`WhatISee` service has been removed from rai in https://github.com/RobotecAI/rai/pull/579. Detections are done in `rai_openset`. Therefore I think the flow is not very realistic